dora/core/client/fs/src/main/java/alluxio/client/file/dora/WorkerLocationPolicy.java

```diff
@@ -1,265 +1,47 @@
-      // check if the worker list has expired
-      long lastUpdateTs = mLastUpdatedTimestamp.get();
-      long currentTs = System.nanoTime();
-      if (currentTs - lastUpdateTs > mWorkerInfoUpdateIntervalNs) {
-        // use CAS to only allow one thread to actually update the timestamp
-        return mLastUpdatedTimestamp.compareAndSet(lastUpdateTs, currentTs);
-      }
-      return false;
-    }
-
-    /**
-     * Lazily initializes the hash ring.
-     * Only one caller gets to initialize the map while all others are blocked.
-     * After the initialization, the map must not be null.
-     */
-    private void maybeInitialize(List<BlockWorkerInfo> workerInfos, int numVirtualNodes) {
-      if (mActiveNodesByConsistentHashing == null) {
-        synchronized (mInitLock) {
-          // only one thread should reach here
-          // test again to skip re-initialization
-          if (mActiveNodesByConsistentHashing == null) {
-            mActiveNodesByConsistentHashing = build(workerInfos, numVirtualNodes);
-            mLastWorkerInfos.set(workerInfos);
-            mLastUpdatedTimestamp.set(System.nanoTime());
-          }
-        }
-      }
-    }
-
-    private boolean hasWorkerListChanged(List<BlockWorkerInfo> workerInfoList,
-                                         List<BlockWorkerInfo> anotherWorkerInfoList) {
-      if (workerInfoList == anotherWorkerInfoList) {
-        return false;
-      }
-      Set<WorkerNetAddress> workerAddressSet = workerInfoList.stream()
-          .map(info -> info.getNetAddress()).collect(Collectors.toSet());
-      Set<WorkerNetAddress> anotherWorkerAddressSet = anotherWorkerInfoList.stream()
-          .map(info -> info.getNetAddress()).collect(Collectors.toSet());
-      return !workerAddressSet.equals(anotherWorkerAddressSet);
-    }
-
-    public List<BlockWorkerInfo> getMultiple(String key, int count) {
-      Set<BlockWorkerInfo> workers = new HashSet<>();
-      int attempts = 0;
-      while (workers.size() < count && attempts < mMaxAttempts) {
-        attempts++;
-        workers.add(get(key, attempts));
-      }
-      return ImmutableList.copyOf(workers);
-    }
-
-    public BlockWorkerInfo get(String key, int index) {
-      NavigableMap<Integer, BlockWorkerInfo> map = mActiveNodesByConsistentHashing;
-      Preconditions.checkState(map != null, "Hash provider is not properly initialized");
-      return get(map, key, index);
-    }
-
-    @VisibleForTesting
-    static BlockWorkerInfo get(NavigableMap<Integer, BlockWorkerInfo> map, String key, int index) {
-      int hashKey = HASH_FUNCTION.hashString(format("%s%d", key, index), UTF_8).asInt();
-      Map.Entry<Integer, BlockWorkerInfo> entry = map.ceilingEntry(hashKey);
-      if (entry != null) {
-        return entry.getValue();
+     * @param conf Alluxio configuration
+     * @return a new instance of {@link BlockLocationPolicy}
+     */
+    // TODO(jiacheng): use enum
```

Comment:
I'm actually thinking how to enable multiple policies here. If a user may customize an implementation, I will put a class name here. Otherwise, I will use an enum. @maobaolong Do you have specific needs for adding your own implementations? If so I will use a class name in this property.

Reply:
@jiacheliu3 Yeah, Class type property is suggested by me to enable open source developer like me can extend a new policy.